homeassistant/components/camera/bomradarcam.py

```diff
@@ -0,0 +1,75 @@
+"""
+Provide animated GIF loops of BOM radar imagery.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/bomradarcam/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.camera import PLATFORM_SCHEMA, Camera
+from homeassistant.helpers import config_validation as cv
+
+REQUIREMENTS = ['bomradarloop==0.1.2']
+
+CONF_DELTA = 'delta'
+CONF_FRAMES = 'frames'
+CONF_ID = 'id'
+CONF_LOC = 'location'
+CONF_NAME = 'name'
+CONF_OUTFILE = 'filename'
+LOGGER = logging.getLogger(__name__)
+
+
+def _validate_schema(config):
+    if config.get('location'):
+        if config.get('id'):
+            raise vol.Invalid("Specify either 'id' or 'location', not both")
+    else:
+        reqs = (config.get('id'), config.get('delta'), config.get('frames'))
+        if not all(reqs):
+            raise vol.Invalid("Specify 'id', 'delta' and 'frames' when"
+                              " 'location' is unspecified")
+    return config
+
+
+PLATFORM_SCHEMA = vol.All(PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DELTA): cv.positive_int,
+    vol.Optional(CONF_OUTFILE): cv.string,
+    vol.Optional(CONF_FRAMES): cv.positive_int,
+    vol.Optional(CONF_ID): cv.string,
+    vol.Optional(CONF_LOC): cv.string,
```

Comment:
Is it possible to validate what locations or ids are allowed?

What happens if a user configures an incorrect location?